### ISSUE TITLE
Document how to view logs when using Minio

### ISSUE THREAD

**Issue by containscafeine** [heptio/ark#387]
I was following the getting started guide, and all the ark related pods are up and running. I am also able to take the backups, but I'm unable to view the logs for the same.

```console
$ ark backup get
NAME      STATUS      CREATED                         EXPIRES   SELECTOR
wp        Failed      2018-03-16 17:20:35 +0530 IST   29d       <none>
wpback    Completed   2018-03-16 17:20:44 +0530 IST   29d       <none>

$ ark backup logs wp
An error occurred: Get http://minio.heptio-ark-server.svc:9000/ark/wp/wp-logs.gz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Content-Sha256=UNSIGNED-PAYLOAD&X-Amz-Credential=minio%!F(MISSING)20180316%!F(MISSING)minio%!F(MISSING)s3%!F(MISSING)aws4_request&X-Amz-Date=20180316T115127Z&X-Amz-Expires=600&X-Amz-SignedHeaders=host&X-Amz-Signature=c8ef8e268775c73d9140937c8019a30bb5718059aabf052d8b4982828365624f: dial tcp: lookup minio.heptio-ark-server.svc on 192.168.0.1:53: no such host
```

**Comment by ncdc** [comment]
If you're using minio, you need to be able to resolve the minio `service`'s cluster DNS entry from wherever you're running your `ark` client. Try `kubectl -n heptio-ark edit config/default` and change the url to `minio.heptio-ark-server.svc.cluster.local`, and then follow these instructions: https://stevesloka.com/2017/06/12/access-minikube-service-from-linux-host/.

**Comment by containscafeine** [comment]
Ah, got it.
I exposed the minio service via NodePort, and then pointed the config at it, works now.
Should this not be documented in the getting started?

**Comment by ncdc** [comment]
Probably a good idea. Let's use this issue for that.

**Comment by ramv76** [comment]
can you please send the steps you followed to resolve this issue? I am getting the same error.

**Comment by ncdc** [comment]
@skriss we might want to add a `publicURL` field for the aws object store config so the urls we send back for logs etc. can be different than the url ark uses to talk to minio.

**Comment by containscafeine** [comment]
@ramv76 FWIW I was running on a single node Kubernetes cluster (minikube), so I'm not sure if the following will work for you, but what I did was -

- change Type to NodePort (https://kubernetes.io/docs/concepts/services-networking/service/#type-nodeport) - https://github.com/heptio/ark/blob/9fc1711d450be8a59ec3293ed75169c3a04751b0/examples/minio/00-minio-deployment.yaml#L61 
- get minio's NodePort from `kubectl get services`
- point s3Url to <node ip>:<node port> at https://github.com/heptio/ark/blob/9fc1711d450be8a59ec3293ed75169c3a04751b0/examples/minio/10-ark-config.yaml#L27

This was a long time back, see if this works for you

**Comment by ramv76** [comment]
Hi, Thank you so much for the updates. Is this possible to configure backup locally in the server itself or like a dedicated backup server to store the backup? if yes, can you please shed some light in this?

**Comment by ramv76** [comment]
Sure, Thank you so much Andy.

**Comment by ramv76** [comment]
@Andy, can you please send me an invite for ramv1010@gmail.com. This will be great help for me. Thanks a lot and sorry for any inconvenience caused in this.

**Comment by ncdc** [comment]
@ramv76 you can go to http://slack.k8s.io and sign up

**Comment by ncdc** [comment]
@sachinar (moving from #871) could you please describe the specific steps you tried that aren't working? Thanks!

**Comment by sachinar** [comment]
I tried with these steps.

`apiVersion: ark.heptio.com/v1
kind: Config
metadata:
  namespace: heptio-ark
  name: default
persistentVolumeProvider:
  name: aws
  config:
    region: minio
backupStorageProvider:
  name: aws
  bucket: ark
  # Uncomment the below line to enable restic integration.
  # The format for resticLocation is <bucket>[/<prefix>],
  # e.g. "my-restic-bucket" or "my-restic-bucket/repos".
  # This MUST be a different bucket than the main Ark bucket
  # specified just above.
  resticLocation: ark/backup-bucket
  config:
    region: minio
    s3ForcePathStyle: "true"
    s3Url: http://minio:9000
`
Here  s3Url I changed to minio.heptio-ark-server.svc.cluster.local:9000, but same error was there.

**Comment by sachinar** [comment]
@ncdc  I tried with 
 s3Url: http://minio.heptio-ark.svc:9000  this also. same error was there.

**Comment by ncdc** [comment]
Please try making the Minio service a NodePort service and set the s3Url to
http://$node:$nodePort

**Comment by sachinar** [comment]
@ncdc Thanks It worked.

**Comment by skriss** [comment]
Closed by #1023
